src/org/opendatakit/briefcase/ui/automation/AutomationPanel.java

```diff
@@ -34,33 +38,76 @@
   private final FormCache formCache;
   private final BriefcasePreferences appPreferences;
 
-  public AutomationPanel(AutomationPanelForm view, TransferForms forms, BriefcasePreferences appPreferences, FormCache formCache) {
+  private Optional<Source<?>> pullSource;
+  private Optional<Source<?>> pushSource;
+
+  public AutomationPanel(AutomationPanelForm view, TransferForms forms, BriefcasePreferences tabPreferences, BriefcasePreferences appPreferences, FormCache formCache) {
     AnnotationProcessor.process(this);
     this.view = view;
     this.forms = forms;
     this.formCache = formCache;
     this.appPreferences = appPreferences;
 
+    // Read prefs and load saved remote server if available
+    this.pullSource = RemoteServer.readPreferences(tabPreferences).flatMap(view::preloadPullSource);
+    this.pullSource.ifPresent(source -> {
+      forms.load(source.getFormList());
+      view.refresh();
+    });
+
+    view.onPullSource(pullSource -> {
+      this.pullSource = Optional.of(pullSource);
+      Source.clearAllPreferences(tabPreferences);
+      pullSource.storePreferences(tabPreferences, getStorePasswordsConsentProperty());
+      forms.load(pullSource.getFormList());
+      view.refresh();
+    });
+
+    view.onPushSource(pushSource -> {
+      Source.clearAllPreferences(tabPreferences);
+      pushSource.storePreferences(tabPreferences, getStorePasswordsConsentProperty());
+      forms.load(pushSource.getFormList());
+      view.refresh();
+    });
+
+    // Read prefs and load saved remote server if available
+    this.pushSource = RemoteServer.readPreferences(tabPreferences).flatMap(view::preloadPushSource);
+    this.pushSource.ifPresent(source -> {
+      forms.load(source.getFormList());
+      view.refresh();
+    });
+
     view.onGenerate(config -> generateScript(isWindows() ? "automation.bat" : "automation.sh", config));
   }
 
   private void generateScript(String scriptName, AutomationConfiguration configuration) {
     Path scriptDirPath = configuration.getScriptLocation().orElseThrow(BriefcaseException::new);
     Path storageDir = appPreferences.getBriefcaseDir().orElseThrow(BriefcaseException::new).getParent();
-    String template = "java -jar {0} --export --form_id {1} --storage_directory {2} --export_directory {3} --export_filename {4}.csv";
+    String exportTemplate = "java -jar {0} --export --form_id {1} --storage_directory {2} --export_directory {3} --export_filename {4}.csv";
 
-    List<String> scriptLines = forms.getSelectedForms()
+    List<String> scriptLines = pullSource.orElseThrow(BriefcaseException::new).pullScriptLines(forms.getSelectedForms(), appPreferences);
```

Comment:
I have some comments about the mechanics that you've laid out here. A `Source` should not know anything about the automation scripts. Any transformation from values in memory to actual script lines should happen here, not in `Source`.

The way I see it, `Source` should return some `Map<String,String>` with its internal params, and then that map could be used to fill in template params here.